src/CoreWCF.RabbitMQ/src/CoreWCF/Channels/RabbitMqConnectionSettings.cs

```diff
@@ -0,0 +1,177 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Linq;
+using System.Net;
+using RabbitMQ.Client;
+
+namespace CoreWCF.Channels
+{
+    public class RabbitMqConnectionSettings
+    {
+        public const string TempQueuePrefix = "corewcf-temp-";
+        private const string AMQPScheme = "amqp";
+        private const string SecureAMQPScheme = "amqps";
+        private const string DefaultVirtualHost = "/";
+
+        public string Host { get; set; }
+        public int Port { get; set; }
+        public string Exchange { get; set; } = string.Empty;
+        public string QueueName { get; set; }
+        public string VirtualHost { get; set; } = DefaultVirtualHost;
+        public virtual string RoutingKey { get; set; }
+        public string UserName { get; set; }
+        public string Password { get; set; }
+        public SslOption SslOption { get; set; }
+        public bool AutomaticRecoveryEnabled => true;
+
+        public static RabbitMqConnectionSettings FromUri(
+            Uri uri,
+            ICredentials credentials = null,
+            SslOption sslOption = null,
+            string virtualHost = DefaultVirtualHost)
+        {
+            sslOption = ConfigureSslOption(sslOption, uri);
+            var queueName = GetQueueNameFromUri(uri);
+            var exchange = GetExchangeFromUri(uri);
+            var routingKey = GetRoutingKeyFromUri(uri, queueName);
+            var userName = SetUserName(uri, credentials);
+            var password = SetPassword(uri, credentials, sslOption);
+
+            return new RabbitMqConnectionSettings
+            {
+                Host = uri.Host,
+                Port = uri.Port,
+                Exchange = exchange,
+                RoutingKey = routingKey,
+                QueueName = queueName,
+                VirtualHost = virtualHost,
+                UserName = userName,
+                Password = password,
+                SslOption = sslOption
+            };
+        }
+
+        public ConnectionFactory GetConnectionFactory()
+        {
+            return new ConnectionFactory
+            {
+                HostName = Host,
+                Port = Port,
+                VirtualHost = VirtualHost,
+                UserName = UserName,
+                Password = Password,
+                Ssl = SslOption,
+                AutomaticRecoveryEnabled = AutomaticRecoveryEnabled
+            };
+        }
+
+        private static string GetQueueNameFromUri(Uri uri)
+        {
+            if (uri.Segments.Length < 2)
+            {
+                throw new ArgumentException($"Expected a non-empty segment in the uri {uri}");
+            }
```

Comment:
Put exception text into string resource. Also should state the expected format of the uri in the exception message.